src/components/map/geometry-item.tsx

```diff
@@ -75,6 +45,19 @@ export const GeometryItem = ({
     );
   }
 
+  const hasDetails = layer.type !== undefined && layer.url && properties?.id;
+  const pointerClassName = hasDetails ? '' : '!cursor-[unset]';
+
+  const featureEventHandler = {
+    click: () => {
+      if (hasDetails) {
+        router.push(
+          `/map/${layer?.type}/${properties?.id}?${params.toString()}`,
+        );
+      }
+    },
+  };
```

Comment:
To avoid an useless attachment event we could check the `hasDetails` before:
```tsx
  const featureEventHandler = {
    ...(hasDetails && {
    click: () => {
        router.push(
          `/map/${layer?.type}/${properties?.id}?${params.toString()}`,
        );
      }
      }),
  };
``` 
But maybe it's less readable...